robottelo/helpers.py

```diff
@@ -208,3 +214,55 @@ def remove_katello_ca(hostname=None):
             )
             if result.return_code != 0:
                 raise AssertionError('Failed to reset the rhsm.conf')
+
+
+def download_server_file(localpath, fileurl):
+    """Downloads a file from server using wget command.
+
+    :param str localpath: local path of file to write to from fileurl.
+    :param str fileurl: A full path to file on server.
+    :return: None
+    :raises: DownloadFileError: If file not downloaded
+
+    """
+    global _YAML_DOWNLOADED
+    if not _YAML_DOWNLOADED:
+        ssh.command('wget -O {0} {1}'.format(localpath, fileurl))
+        _YAML_DOWNLOADED = True
+    if not os.path.exists(localpath):
+        raise DownloadFileError('Failed to download a file from {0}'.format(
+            fileurl))
+
+
+def get_all_yaml_data():
+    """Downloads and fetches all data from YAML file.
+
+    :return: Dict type contains data fetched from yaml in the form of
+    {node:{subnode:{fields:data}}}
+
+    """
+    yaml_path = '/tmp/preUpgrade_data.yml'
```

Comment:
A temporary file should be created instead of a hardcoded path. Check the temporary file Python module